modyn/trainer_server/internal/dataset/key_sources/local_key_source.py

```diff
@@ -0,0 +1,62 @@
+import os
+from typing import Optional
+
+from modyn.common.trigger_sample import TriggerSampleStorage
+from modyn.trainer_server.internal.dataset.key_sources import AbstractKeySource
+
+LOCAL_STORAGE_FOLDER = "/tmp/.tmp_offline_dataset"
+
+
+class LocalKeySource(AbstractKeySource):
+    def __init__(self, pipeline_id: int, trigger_id: int) -> None:
+        super().__init__(pipeline_id, trigger_id)
+
+        self._trigger_sample_storage = TriggerSampleStorage(LOCAL_STORAGE_FOLDER)
+
+    def get_keys_and_weights(self, worker_id: int, partition_id: int) -> tuple[list[int], Optional[list[float]]]:
+        path = self._trigger_sample_storage._get_file_path(self._pipeline_id, self._trigger_id, partition_id, worker_id)
```

Comment:
Yes, indeed, it was not the most elegant solution. I modified the function so that it automatically puts the extension (.npy) so the path does not have to be touched.